### PR TITLE
mgmt: mcumgr: lib: fs: Add file read/write access callback

### PR DESCRIPTION
**Pull request by nordicjm**
Adds a hook that applications can register for and then use to receive callbacks that can grant or deny access requests to files via the fs mcumgr system.

    doc: release-notes: Add mcumgr file read/write request hook details
    
    Adds the file read/write request hook that applications can use to allow
    or decline requests to files to the release notes.
    
    doc: guides: device_mgmt: smp: fs: Add fs hook details
    
    Adds details about the file read/write request hook that applications
    can use to allow or decline requests to files.
    
    mgmt: mcumgr: lib: fs: Add file read/write access callback
    
    This allows an application to inspect a mcumgr file access command and
    either allow it or deny it with a result code.

### CODE REVIEW

File: doc/services/device_mgmt/smp_groups/smp_group_8.rst [hunk 1]
```diff
@@ -37,6 +37,13 @@ belonging to specified download session. This means that the file is not
 locked in any way or exclusively owned by mcumgr, for the time of download
 session, and may change between requests or even be removed.
 
+.. note::
+    By default, all file upload requests are unconditionally allowed, however if
+    the Kconfig option `FS_MGMT_FILE_ACCESS_HOOK` is enabled, than an
+    application can register a handler for a `fs_mgmt_on_evt_cb` callback which
+    can be used to allow or decline access to reading from/writing to a
+    particular file.
```
Comment (mbolivar-nordic) [inline doc/services/device_mgmt/smp_groups/smp_group_8.rst:45]:
```suggestion
    By default, all file upload requests are unconditionally allowed. However, if
    the Kconfig option :kconfig:option:`FS_MGMT_FILE_ACCESS_HOOK` is enabled,
    then an application can register a handler for a ``fs_mgmt_on_evt_cb`` callback. This
    can be used to allow or decline access to reading from/writing to a
    particular file.
```

File: doc/services/device_mgmt/smp_groups/smp_group_8.rst [hunk 2]
```diff
@@ -144,6 +151,13 @@ session, and may change between requests or even be removed.
     of bool flag indicating in-progress upload, last successfully uploaded offset
     and total length only.
 
+.. note::
```
Comment (mbolivar-nordic) [inline doc/services/device_mgmt/smp_groups/smp_group_8.rst:154]: Similar comments here

**Review by mbolivar-nordic — COMMENTED** [review]
The idea seems reasonable, but I don't know the area so well. I have a couple of suggestions for the docs.

File: doc/services/device_mgmt/smp_groups/smp_group_8.rst (hunk 1 above)
Comment (lairdjm) [reply doc/services/device_mgmt/smp_groups/smp_group_8.rst]: Fixed

File: doc/services/device_mgmt/smp_groups/smp_group_8.rst (hunk 2 above)
Comment (lairdjm) [reply doc/services/device_mgmt/smp_groups/smp_group_8.rst:155]: Fixed

**Review by de-nordic — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]
LGTM

**Pull request merged**